src/monocular.cpp

```diff
@@ -196,12 +203,28 @@ MapMsg MonocularSlamNode::MapPointsToPointCloud (std::vector<ORB_SLAM3::MapPoint
             data_array[0] = map_pt[0];
             data_array[1] = map_pt[1];
             data_array[2] = map_pt[2];
-
-            memcpy(cloud_data_ptr + (i * cloud.point_step), data_array, num_channels * sizeof(float));
```

Comment:
can you help me understand where the memory for the point cloud is being allocated?

Reply:
You can find the memory allocation here: https://github.com/shandilya1998/ros2_orb_slam3/blob/b20237b9d8d3e0e92c8fce291865567fdbe37117/src/monocular.cpp#L187

Reply:
cloud itself is unassigned right? 
`MapMsg::UniquePtr msg = std::make_unique<MapMsg>();`

Reply:
memory is allocated to the cloud, but it is empty. I wanted to point to the data array without having to use the complex notation.